### PR TITLE
#162948859 Implement tracking of comment history

### PR DESCRIPTION
**Pull request by d-kahara**
#### What does this PR do?
Allows users to track history of comments they make on the platform after editing

#### Description of Task to be completed?
Authenticated users should be able to track edits they make to a particular comment.

#### How should this be manually tested?

- Clone the repo locally
- cd ah-django-unchained
- git checkout ft-track-comment-history-162948859
- get started with the app as stated in the README section
- to add an article ensure you obtain a token from logging in or signing up if not signed up
- Using the `POST api/articles/` endpoint, create an article
- Using the ` POST api/articles/<slug>/comments/` endpoint, add a comment to the article where `<slug>` is the slug generated after the article is posted 
- Using the `PUT api/articles/<slug>/comments/<id>/` endpoint,  update the comment you created where `<id>` is the id of the comment
- Using the ` GET api/articles/<slug>/comments/<id>/history/` endpoint, view the history of the comment you created then updated.
#### Screenshots
image

#### Relevant Pivotal Tracker Stories

[#162948859](https://www.pivotaltracker.com/story/show/162948859)

### CODE REVIEW

File: authors/apps/articles/tests/test_comment_history.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        """Sign up user and get token"""
+        register = self.client.post(self.signup_url,
+                                    self.signup_data,
+                                    format='json')
+
+        token = register.data["token"]
+        return token
+
+    def create_article(self, article, token):
+        del article['images']
```
Comment (kwameasiago) [inline authors/apps/articles/tests/test_comment_history.py]: No need of deleting image, just remove it from the create_article_data

**Review by kwameasiago — APPROVED** [review]
Looks good

**Comment by ignatiusukwuoma** [comment]
You might need to run `python manage.py makemigrations --merge` to merge the migration files

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -677,3 +677,29 @@ def get_all_comments(slug, id):
...
+
+    def get(self, request, id, slug, **kwargs):
+
+        try:
+            # get an article that matches the slug specified
+            # and return a message if the article does not exist
+            Article.objects.get(slug=slug, is_deleted=False)
+        except ObjectDoesNotExist:
+            raise NotFound("This article does not exist")
+        id = self.kwargs['id']
```
Comment (ignatiusukwuoma) [inline authors/apps/articles/views.py]: is this the same id with the one passed into the function? if it is, is this still required?

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -677,3 +677,29 @@ def get_all_comments(slug, id):
...
+        edited_comment = Comment.history.filter(id=id, is_deleted=False)
+        if not edited_comment:
+            message = {'error': 'No edit history for this comment'}
```
Comment (ignatiusukwuoma) [inline authors/apps/articles/views.py]: error message should be that the comment doesn't exist because even if only one comment was made, it'll be shown.

**Review by ignatiusukwuoma — APPROVED** [review]

**Pull request merged**